Tasks/JavaToolInstallerV0/javatoolinstaller.ts

```diff
@@ -55,19 +47,78 @@ async function getJava(versionSpec: string) {
         const extractSource = buildFilePath(extractLocation, compressedFileExtension, fileNameAndPath);
         const javaFilesExtractor = new JavaFilesExtractor();
         jdkDirectory = await javaFilesExtractor.unzipJavaDownload(extractSource, compressedFileExtension, extractLocation);
-    } else { //JDK is in a local directory. Extract to specified target directory.
+    } else if ('LocalDirectory' == taskLib.getInput('jdkSourceOption', true)) { //JDK is in a local directory. Extract to specified target directory.
         console.log(taskLib.loc('RetrievingJdkFromLocalPath'));
+        cleanDestinationDirectory(cleanDestDir, extractLocation);
+
         compressedFileExtension = getFileEnding(taskLib.getInput('jdkFile', true));
         const javaFilesExtractor = new JavaFilesExtractor();
         jdkDirectory = await javaFilesExtractor.unzipJavaDownload(taskLib.getInput('jdkFile', true), compressedFileExtension, extractLocation);
+    } else if ('AptGet' == taskLib.getInput('jdkSourceOption', true)) { //Install JDK via apt-get
+        //Check that it is Linux. Apt-get does not work on all Linux distributions, however we have no way of checking which one is being used.
+        if(os.platform() === 'linux') {
+            let escapedScript = path.join(__dirname, 'aptGetJavaInstall.sh').replace(/'/g, "''");
+            fs.chmodSync(escapedScript, "777");
+
+            let scriptRunner = taskLib.tool(taskLib.which('bash', true))
+            .arg('--noprofile')
+            .arg(`--norc`)
+            .arg(escapedScript)
+            .arg(`${versionSpec}`);
+                
+            let result: tr.IExecSyncResult = scriptRunner.execSync();
+            if (result.code != 0) {
+                throw taskLib.loc("AptGetFailed", result.error ? result.error.message : result.stderr);
+            }
+
+            setEnvironmentVars(versionSpec, '/usr/lib/jvm/java-' + versionSpec + '-openjdk-amd64');
+        } else {
+            console.log(taskLib.loc('NotLinuxPlatform'));
+            console.log(taskLib.loc('CurrentPlatformInfo', os.platform()));
+        }
+    } else { //Install JDK via chocolatey command
+        //check that it is Windows
+        if(os.platform() === 'win32') {
+            let filePath = path.join(__dirname, 'chocoJavaInstall.ps1').replace("'", "''");
+            let command = `. '${filePath}' -VersionSpec '${versionSpec}'`;
+            let powershell = taskLib.tool(taskLib.which('pwsh') || taskLib.which('powershell') || taskLib.which('pwsh', true))
+                .arg('-NoLogo')
+                .arg('-NoProfile')
+                .arg('-NonInteractive')
+                .arg('-Command')
+                .arg(command);
+            const options: tr.IExecOptions = <any>{
```

Comment:
asserting `<any>` then declaring `IExecOptions` doesn't seem right.  Why does the object not type match  `IExecOptions` ?